docs/blueprints/referral-manager-v1.md

```diff
@@ -160,77 +156,45 @@ func Redeem(ctx, token, userID, satelliteURL) error {
-
-    return eligibleUsersCounts, nil
-}
 
 // GenerateTokens generates tokens, saves those in the referral manager db
-GenerateTokens(tokensPerUser int) (int, error) {
-    eligibleUsers := referralManager.eligibleUsers
+GenerateTokens(tokensPerUser int, satelliteURLs []string, dryRun bool) (tokenCount, eligibleUserCount int, error) {
+    eligibleUsers := db.GetEligibleUsers(satelliteURLs, tokensPerUser)
     if eligibleUsers == nil {
-        return nil, Error.New("No users to generate tokens for. Make sure to run GetEligibleUsers first")
+        return nil, Error.New("No users to generate tokens for.")
     }
-    var tokens []Token
-    for _, users := range eligibleUsers {
-        for i:=0; i<tokensPerUser; i++ {
-            newToken := Token{data: generateRandomToken(), user: user.ID, satellite: user.satelliteURL}
-            db.CreateToken(newToken)
-            tokens = append(tokens, newToken)
+    newTokenCount := 0
+    eligibleUserCount := len(eligibleUsers)
+    for _, user := range eligibleUsers {
+        for i:=user.NewTokens; i<tokensPerUser; i++ {
+            newTokenCount++
         }
     }
 
-    return len(tokens), nil
-}
+    if !dryRun {
+        db.UpdateUserNewTokensBatch(eligibleUsers, tokensPerUser)
+    }
 
-// ClearUsers sets eligibleUsers to be empty
-func ClearUsers(ctx) {
-    referralManager.eligibleUsers = nil
-    return nil
+    return newTokenCount, eligibleUserCount, nil
 }
 ```
 
 **CLI code:**
 ```
 startCmd() {
-    eligibleUsers, err := referralManager.GetEligibleUsers()
-    if err == UsersAlreadyRetrievedErr {
-        confirm := prompt("Referral token generation already in progress or canceled early. Do you want to continue anyway?")
-        if confirm != "y" {
-            exit
-        }
-        referralManager.ClearUsers()
-    } else if err != nil {
-        exit
-    }
-
-    fmt.Printf("Total available user counts: %d\n", len(eligibleUsers))
-
-    tokensPerUser := prompt("How many tokens do you want to generate per user?\n")
-    totalTokens := tokensPerUser * total
-    confirm := prompt("Are you sure you want to generate %d total tokens?\n", totalTokens)
+    tokenCount, userCount := referralManager.GenerateTokens(tokensPerUser, args.SatelliteURLs, flags.dryRun)
 
-    if confirm != "y" {
-        referralManager.ClearUsers()
-        exit
+    fmt.Printf("Successfully created %d tokens for %d users.\n", tokenCount, userCount)
+    if flags.dryRun {
+        fmt.Println("This was a dry run. Run again without the --dry-run flag to actually generate tokens.")
     }
-
-    newTokens := referralManager.GenerateTokens(tokensPerUser)
-
-    fmt.Printf("Successfully created %d tokens.\n", len(newTokens))
 }
 ```
 
 ## Wrapup
 
 Team Green will be responsible implementing this blueprint.
 
-## Open issues
\ No newline at end of file
+## Open issues
+
+Should we have another CLI command for querying the global number of unredeemed tokens + ungenerated tokens?
```

Comment:
I think we would be fine without it since the data can be accessed through redash